storage/innobase/log/log0log.cc

```diff
@@ -1279,7 +1279,8 @@ It just need to flush the file header block with current master key.
 bool
 log_write_encryption(
 	byte*	key,
-	byte*	iv)
+	byte*	iv,
+	redo_log_encrypt_enum redo_log_encrypt)
```

Comment:
Please document this  parameter in the doxygen comments.

Also why do we have this variable when there is global variable srv_redo_log_encrypt?

Reply:
Because this function can be called before srv_redo_log_encryption is set. This was a change introduced in 8.0.16, where it's a non-issue (it can only be on and off, and this method is only called when it's on).

Reply:
Confused here. How can log_write_encryption() be called  before if srv_red_log_encryption is  SET?

Reply:
That's how upstream refactored the setter: now when the user changes the redo_log_encryption setting, first the initialization is completed (including writing out the encryption header), and after that, the variable is changed.